harness/determined/util.py

```diff
@@ -26,7 +26,9 @@ def download_gcs_blob_with_backoff(blob: Any, n_retries: int = 32, max_backoff:
 
 
 def is_overridden(full_method: Any, parent_class: Any) -> bool:
-    return cast(bool, full_method.__qualname__.partition(".")[0] != parent_class.__name__)
+    if callable(full_method):
```

Comment:
question: What does this change do?

Reply:
This is fix for that users can't set their optimizer by `self.optimizer=...`. We check if `self.optimizer` is overridden by checking `__qualname__` however, only functions have that attribute.

Reply:
Ah.  That's tricky.  We should explain that in a comment at a minimum.

Reply:
Done